packages/upset/src/components/ElementView/ElementVisualization.tsx

```diff
@@ -1,36 +1,99 @@
+import {
+  useContext, useMemo, useRef, useState,
+} from 'react';
+import { SignalListener, VegaLite } from 'react-vega';
 import { useRecoilValue } from 'recoil';
 
-import { bookmarkedIntersectionSelector } from '../../atoms/config/currentIntersectionAtom';
+import { elementSelectionToBookmark, elementSelectionsEqual, isElementSelection } from '@visdesignlab/upset2-core';
+import { Button } from '@mui/material';
+import { bookmarkSelector, elementColorSelector } from '../../atoms/config/currentIntersectionAtom';
 import { histogramSelector, scatterplotsSelector } from '../../atoms/config/plotAtoms';
-import { elementItemMapSelector } from '../../atoms/elementsSelectors';
+import { elementItemMapSelector, selectedElementSelector } from '../../atoms/elementsSelectors';
 import { AddPlotDialog } from './AddPlotDialog';
 import { generateVega } from './generatePlotSpec';
+import { ProvenanceContext } from '../Root';
+import { UpsetActions } from '../../provenance';
 
+/**
+ * Displays a matrix of plots representing the elements in the current intersection selection & bookmarks.
+ * @returns
+ */
 export const ElementVisualization = () => {
+  /**
+   * State hooks
+   */
+
   const [openAddPlot, setOpenAddPlot] = useState(false);
   const scatterplots = useRecoilValue(scatterplotsSelector);
   const histograms = useRecoilValue(histogramSelector);
-  const bookmarked = useRecoilValue(bookmarkedIntersectionSelector);
+  const bookmarked = useRecoilValue(bookmarkSelector);
   const items = useRecoilValue(elementItemMapSelector(bookmarked.map((b) => b.id)));
+  const currentSelection = useRecoilValue(selectedElementSelector);
+  const selectColor = useRecoilValue(elementColorSelector);
+  // This will default to the savedSelection because brushHandler fires on the default selection in generateVega()
+  const { actions }: {actions: UpsetActions} = useContext(ProvenanceContext);
 
+  /**
+   * Hooks
+   */
+
+  const draftSelection = useRef(currentSelection?.selection);
+  const vegaSpec = useMemo(
+    () => generateVega(scatterplots, histograms, selectColor, currentSelection?.selection),
+    [scatterplots, histograms, selectColor, currentSelection?.selection],
+  );
+
+  /**
+   * Functions
+   */
+
+  /**
+   * Closes the AddPlotDialog
+   */
   const onClose = () => setOpenAddPlot(false);
 
+  /**
+   * Saves brush bounds to state when the interactive brush is used.
+   * @param {string}  _     Name of the signal: "brush"
+   * @param {unknown} value Should be an object mapping the names of the attributes being brushed over
+   *  to an array of the bounds of the brush, but Vega's output format can change if the spec changes.
+   */
+  const brushHandler: SignalListener = (_: string, value: unknown) => {
+    if (!isElementSelection(value)) return;
+    draftSelection.current = value;
+  };
+
   return (
-    <Box>
+    <Box
+      onClick={() => {
+        // Since onClick fires onMouseUp, this is a great time to save
```

Comment:
Should we not just explicitly use `onMouseUp`?

Reply:
`onMouseUp` doesn't bubble up from `<Vegalite>`, I'm updating the comment

Reply:
Addressed in 9928adf33c7ec8cc57445051fac0db037439fb9d